File: languages/tree-sitter-stack-graphs-java/test/statement/for_statement.java
class A {
  void f() {
    boolean x;
    for(x; x; x) {
      //^ defined: 3
      //   ^ defined: 3
      //      ^ defined: 3
        f();
      //^ defined: 2
    }
  }
  void g() {
    for(boolean x; x; x) {
      //           ^ defined: 13
      //              ^ defined: 13
        g();
      //^ defined: 12
    }
  }

  void h() {
    int k = 5;
    for (int i = 0, j = i; i < 123; k++) {
                     // ^ defined: 23
                                 // ^ defined: 22
    }
  }
}

File: languages/tree-sitter-stack-graphs-javascript/test/statements/for_statement.js
let x = 1;

// Flow in

for (let y = (x, 1);
    //        ^ defined: 1
    /**************/ x < y;
    //               ^ defined: 1
    //                   ^ defined: 5, 10
    /*********************/ y = (x, y)) {
    //                           ^ defined: 1
    //                              ^ defined: 5, 10
    /**/ x;
    //   ^ defined: 1
    /**/ y;
    //   ^ defined: 5, 10
    z = 1;
}

// Flow out

/**/ y;
//   ^ defined: 5, 10

/**/ z;
//   ^ defined: 17

// Flow around

/**/ x;
//   ^ defined: 1